lib/file-operations.js

```diff
@@ -0,0 +1,91 @@
+const fs = require("fs");
+const { access, constants } = require ('node:fs/promises');
+function makeDir(folder){
+return new Promise((resolve,reject)=>{
+    fs.mkdir(folder,(err)=>{
+        if(err){
+            resolve({success: false,error : err})
+        }
+        else {
+            resolve({success: true })
+        }
+    })
+})
+}
+
+function deleteFile(file){
+    return new Promise((resolve, reject)=>{
+        fs.unlink(file,(err)=>{
+            if(err){
+                resolve({success:false,error : err})
+            }
+            else {
+                resolve({success : true})
+            }
+        })
+    })
+}
+
+function copyFile(source,dest){
+    return new Promise((resolve)=>{
+        fs.copyFile(source,dest,(err)=>{
+            if(err){
+                resolve({success : false,error:err})
+            }
+            else {
+                resolve({success : true})
+            }
+        })
+    })
+}
+
+function moveFile(source,dest){
+    return new Promise(resolve => {
+        fs.rename(source,dest,(err)=>{
+            if(err){
+                resolve({success : false, error : err})
+            }
+            else {
+                resolve({success : true})
+            }
+        })
+    })
+}
+
+function deleteFolder(path){
+    return new Promise(resolve => {
+        fs.rmdir(path,(err)=>{
+            if (err){
+                resolve({success : false, error : err})
+            }
+            else {
+                resolve({success : true})
+            }
+        })
+    })
+
+}
+
+async function checkExists(path){
+    try {
+        await access(path,constants.R_OK)
+        return true
+    }
+    catch (error){
+        //todo log
+        return false
+    }
+}
+        
+const fileFuncs = {
+    makeDir : makeDir,
+    deleteFile : deleteFile,
+    deleteFolder : deleteFolder,
+    moveFile : moveFile,
+    copyFile : copyFile,
+    checkExists : checkExists
+
+}
+module.exports = {
+   fileFuncs
+}
```

Comment:
y not export it directly?

also, no need to write ":" if the key string and the variable which goes in the value are same

so this can be change to 
```
module.exports = {
    makeDir,
    deleteFile,
    deleteFolder,
    moveFile,
    copyFile,
    checkExists
}

Reply:
yes didn't knew one can do that

Reply:
done